cmd/generate-catalog/generate.go

```diff
@@ -0,0 +1,279 @@
+package main
+
+import (
+	// needed for digest algorithm validation
+	_ "crypto/sha256"
+
+	"encoding/base64"
+	"fmt"
+	"log"
+	"os"
+	"slices"
+
+	semver "github.com/Masterminds/semver/v3"
+	"github.com/distribution/reference"
+	"github.com/goccy/go-yaml"
+)
+
+const (
+	inputFile                       = "bundles.yaml"
+	outputFile                      = "catalog-template.yaml"
+	iconFile                        = "icon.png"
+	firstLineHeadComment            = "DO NOT EDIT. This file is generated by cmd/generate-catalog/generate.go script."
+	secondLineHeadComment           = "Any manual changes will be overwritten."
+	channelDeprecationMessage       = "This version is no longer supported. Please switch to the `stable` channel or a channel for a version that is still supported. Find supported versions in RHACS support policy document: https://access.redhat.com/support/policy/updates/rhacs \n"
+	bundleDeprecationMessage        = "This Operator version is no longer supported. Use a later version that is supported. Find supported versions in the RHACS support matrix: https://access.redhat.com/support/policy/updates/rhacs \n"
+	bundleBrokenMessage             = "This Operator version is broken and should not be used. Use a later version that is supported.  Find supported versions in the RHACS support matrix: https://access.redhat.com/support/policy/updates/rhacs \n"
```

Comment:
[nit] It's not a support matrix
```suggestion
	bundleDeprecationMessage        = "This Operator version is no longer supported. Use a later version that is supported. Find supported versions in the RHACS support policy document: https://access.redhat.com/support/policy/updates/rhacs \n"
	bundleBrokenMessage             = "This Operator version is broken and should not be used. Use a later version that is supported.  Find supported versions in the RHACS support policy document: https://access.redhat.com/support/policy/updates/rhacs \n"
```